src/domain/flow-system/services/src/flow_configuration/flow_configuration_service_impl.rs

```diff
@@ -34,17 +35,30 @@ pub struct FlowConfigurationServiceImpl {
 #[scope(Singleton)]
 impl FlowConfigurationServiceImpl {
     pub fn new(
-        event_store: Arc<dyn FlowConfigurationEventStore>,
         time_source: Arc<dyn SystemTimeSource>,
         event_bus: Arc<EventBus>,
+        catalog: Catalog,
     ) -> Self {
         Self {
-            event_store,
             time_source,
             event_bus,
+            catalog,
         }
     }
 
+    async fn try_load_configuration(
+        &self,
+        flow_key: FlowKey,
+    ) -> Result<Option<FlowConfiguration>, TryLoadError<FlowConfigurationState>> {
+        DatabaseTransactionRunner::run_transactional_with(
+            &self.catalog,
+            |event_store: Arc<dyn FlowConfigurationEventStore>| async move {
+                FlowConfiguration::try_load(flow_key, event_store.as_ref()).await
+            },
+        )
+        .await
+    }
```

Comment:
Almost all operations of `FlowConfigurationService` are called only from GraphQL, except `list_enabled_configurations`. Those queries already run in a transaction created by HTTP middleware.

I think a service like `FlowConfigurationService` should not be concerned of transactions management at all. I think transactions should be controlled by 2 types of places:
 - middleware for HTTP queries
 - actively running components with their own control loop (`TaskExecutor`, `FlowService`)
 
 
 ```
 
     pub async fn run(self) -> Result<(), InternalError> {
        tokio::select! {
            res = self.server => { res.int_err() },
            res = self.task_executor.run() => { res.int_err() },
            res = self.flow_service.run(self.time_source.now()) => { res.int_err() }
        }
    }
    
    ```

Reply:
`list_enabled_configurations` is only used at startup time in `FlowService`, which should be responsible for transaction management, as an active component.

Reply:
> I think a service like `FlowConfigurationService` should not be concerned of transactions management at all

Agree, done -- except for `AsyncEventHandler<DatasetEventDeleted>`